### ISSUE TITLE
jedi-vim doesn't search pydoc3

### ISSUE THREAD

**Issue by GianniGi** [davidhalter/jedi-vim#982]
### Issue

shift-k doesn't search pydoc3

### Steps to reproduce
with the cursor on a word, for example on "while", press shift-k
it says "jedi-vim: no documentation found for that"
I'd like to configure jedi-vim to open pydoc3 and to search on all the python standard documentation

**Comment by davidhalter** [comment]
This should be improved with Jedi master if we start using `jedi.Script().help()`. So just a matter of time.

**Comment by GianniGi** [comment]
thanks, meanwhile i've added this to vimrc

nnoremap <buffer> <F12> :<C-u>execute "!pydoc3 " . expand("<cword>")<CR>

I've discovered that jedi-vim is very helpful because searches also inside the documentation of the imported modules, so thanks to all the developers for your precious work.

**Comment by GianniGi** [comment]
And yet I don't understand why searching, for example, for "pop", yelds no result on pydoc or jedi...

**Comment by GianniGi** [comment]
sincerely configuring
nnoremap :execute "!pydoc3 " . expand("")
having different systems, operating systems, and python version is such a mess that I don't know why you don't include it in your plugin. It would be very valuable, furthermore because you need to update the systems sometimes, and then you have to fix the vimrc

**Comment by davidhalter** [comment]
... I'm pretty annoyed by your behavior. I don't know how new you are to open source and in general to working with people in the internet. You might be very new to this and if that's the case I'm sorry. I'm a bit pissed, but I will definitely calm down again :). Here are a few pointers:

1. If you open an issue, please try to write the issue in the way other people write it or at least don't delete stuff that was there to be filled out. This is absolute basics. If you don't do this you basically already disqualify.
2. Take a few minutes when writing a comment. Reread it. Add punctuation and write proper English if possible. Your comments feel very sloppy. I usually don't care, because I'm myself guilty of doing this, but you seem to be on another level. Especially, because you seem to want people to do things for you that you don't pay. Your comments don't value our time at all. I also don't care if English is not your first language. We have a lot of people here that are not great writers (like me), but at least they try hard to write a good issue.
3. The time I take to answer you in general, could just as well be spent with my girlfriend or writing code or playing guitar. Something I like doing. I don't like answering and debugging crappy issues. We  maintainers usually don't like the "support" part of our work, we just want to create a good polished product and helping you guys here is by far not my favorite work.

You repeatedly ignored step one, so in my opinion you already disqualify. But usually people don't try again when I repeatedly close their issues. So please take a few minutes think about what you might have done wrong and then write better issues in the future.

-----

> having different systems, operating systems, and python version is such a mess that I don't know why you don't include it in your plugin.

This will definitely not be part of jedi-vim, because it's already part of Jedi master. As I said before: *This is just a matter of time*.

**Comment by GianniGi** [comment]
You don't want to include that feature by principle, that's why in all
these years it hasn't been implemented. It's your software, so you're free
to do as you want.
I'm not annoyed by this fact, but I'm just upset with myself because I've
tried to use a plugin that doesn't work with my vimrc. The more I configure
it, the more buggy and unusable it becomes.

Il giorno giovedì 16 gennaio 2020, Dave Halter <notifications@github.com>
ha scritto:

**Comment by davidhalter** [comment]
Lol. You really don't want to understand, haha. It's not a principle. It's that Jedi has support for what you are asking for already. It just needs to get here :)

**Comment by blueyed** [comment]
@GianniGi 
I agree with @davidhalter in general.

As for:

> nnoremap :execute "!pydoc3 " . expand("")

jedi-vim is not a collection of "Python mappings/config", but a wrapper around Jedi mostly.
Apart from that `expand("")` is empy (i.e. it does not work).
And it is easy to configure yourself.  I use `keywordprg=pydoc` btw, `<leader>K` for `:Unite ref/pydoc`, and `<leader>_K` for jedi-vim's `jedi#show_documentation`.

From what I can tell Jedi master's documentation is good already in general, and if you want to help you could rather follow up on:

> This should be improved with Jedi master if we start using `jedi.Script().help()`. So just a matter of time.

**Comment by GianniGi** [comment]
> Lol. You really don't want to understand, haha. It's not a principle. It's that Jedi has support for what you are asking for already. It just needs to get here :)

I don't know why you're particularly interested in this topic. This feud against pydoc has been going on for at least 7 years. Not that I care about this, but I just wanted to give an example. That's at least 7 years that you spent giving users a limited experience and boring answers just because you must follow a philosophy. There's nothing wrong with that, but you can't be upset if people don't want to embrace your thougths. All while being obviously thankful for your plugin.

**Comment by GianniGi** [comment]
> As for:
> 
> > nnoremap :execute "!pydoc3 " . expand("")

Absolutely wrong. It's not as simple as that. If you want to include a method or a class, you have to include another mapping. And more mappings if you have different computers and different OSes. Still I haven't been able to make it work everywhere.

**Comment by blueyed** [comment]
@GianniGi 
Yeah, so it is not simply "doing it just like that".
I suggest you use `keywordprg` for now, which then would work also on visual selections, e.g. with `str.split`.
Jedi knows more about the identifier at the cursor position, and therefore could better invoke "pydoc", but I agree that it should rather use its own method then.

@davidhalter 
What do you think about a helper function that would return the identifier at the cursor position, so that it could be used in a mapping?

**Comment by davidhalter** [comment]
> What do you think about a helper function that would return the identifier at the cursor position, so that it could be used in a mapping?

What's the reason for that and what would it return?

> This feud against pydoc has been going on for at least 7 years. Not that I care about this, but I just wanted to give an example. That's at least 7 years that you spent giving users a limited experience and boring answers just because you must follow a philosophy.

This is simply not true. If you were around like 5 years ago, you would know that at that time, pydoc was part of Jedi's `goto_definitions`, which caused bugs, so we removed it again. Then I implemented pydoc functionality __within Jedi__. A few weeks later you come along to ask us to work with pydoc. This is the first time somebody complained about pydoc not being there. So all in all it has been 6 days of this "war against pydoc", which is not really a war, because pydoc is used within Jedi at the moment. It just takes 2 more weeks or maybe a month until Jedi is released and we upgrade it here.

Now I ask you: Does it make sense to know spend a day working on pydoc when I have to undo this work in two weeks? Not really IMO.
